lib/framework/procbot.ts

```diff
@@ -45,34 +44,31 @@ interface ServiceMap<T> {
  * * Retrieve configuration file from either the local FS or a ServiceEmitter
  */
 export class ProcBot {
+	/**
+	 * Retrieve the binary path for the Node dependencies.
+	 * @returns Promise containing a string denoting the absolute path.
+	 */
+	public getNodeBinPath = _.memoize((): Promise<string> => {
```

Comment:
I see what you're trying to do here, but [the lodash docs](https://lodash.com/docs/4.17.4#memoize) state that it will use the parameter as a key into an array. I'm not sure what will happen here, so I'd be wary at least.